Azure Functions .dacpac Resolution Fix

varon opened the pull request:
## Proposed Changes

This simply searches a much wider set of locations to try and find a .dacpac. Doing some path magic (and it's best effort) to try and ensure that the path is resolved in wider circumstances.

Specifically, this solves the case where EntryAssembly points to a totally different location, such as when running under the Azure Functions harness.

## Types of changes

What types of changes does your code introduce to SQLProvider?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)


## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Build and tests pass locally
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

Thorium commented:
Looks generally good to me. 
Just a question, should it somehow try-catch the ".." as there could be some possible issues, e.g.:

- What if there is no user permissions to go folders going parent of the root, ".." ?
- What if you have mapped the path to be a root (like Windows map network drive) and ".." doesn't exist?

varon commented:
Integrated all of your suggestions:
- Reduced path duplication
- Added a try-catch around the file opening (for permission issues)
- Handled badly formed paths by just replacing with '.'
- Also added a (likely the most useful) environment variable: `SQLPROVIDER_SSDT_DACPAC_FILE_LOOKUP_PATH`

Did some testing on network paths and local too. This should also work as expected.

Thorium commented:
Thanks, released in NuGet version 1.2.8

varon commented:
You absolute legend. Thank you!
